tests/components/lacrosse_view/test_config_flow.py

```diff
@@ -161,3 +163,148 @@ async def test_form_unexpected_error(hass: HomeAssistant) -> None:
 
     assert result2["type"] == FlowResultType.FORM
     assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_already_configured_device(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch("lacrosse_view.LaCrosse.login", return_value=True,), patch(
+        "lacrosse_view.LaCrosse.get_locations",
+        return_value=[Location(id=1, name="Test")],
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["step_id"] == "location"
+    assert result2["errors"] is None
+
+    with patch(
+        "homeassistant.components.lacrosse_view.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {
+                "location": "1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == FlowResultType.CREATE_ENTRY
```

Comment:
Create a `MockConfigEntry` instead and add it to Home Assistant and then start the flow that should abort.